src/sync.rs

```diff
@@ -258,8 +258,8 @@ impl Context {
         for item in &items.items {
             match &item.data {
                 SyncDataOrUnknown::SyncData(data) => match data {
-                    AddQrToken(token) => self.add_qr_token(token).await,
-                    DeleteQrToken(token) => self.delete_qr_token(token).await,
+                    AddQrToken(token) => self.add_qr_token(token, item.timestamp).await,
```

Comment:
This is a remote timestamp, so better to limit it like `MimeMessage::get_timestamp_sent()` does, maybe even w/o any tolerance